crates/autopilot/src/run.rs

```diff
@@ -538,8 +538,13 @@ pub async fn run(args: Arguments, shutdown_controller: ShutdownController) {
     );
 
     let liveness = Arc::new(Liveness::new(args.max_auction_age));
-    let readiness = Arc::new(Some(AtomicBool::new(false)));
-    observe::metrics::serve_metrics(liveness.clone(), args.metrics_address, readiness.clone());
+    let startup = Arc::new(Some(AtomicBool::new(false)));
```

Comment:
Is this ever `None`? Seems a bit overkill to wrap an atomic bool with an arc

And as a follow up, what is `None` supposed to represent?

Alternatively, could it be better to use a separate enum to write this and wrap that in the arc instead? That way we get explicit states with the name etc

Reply:
This follows the idea of the original proposal, that if one does not want to use the startup probe feature, then it would be None. It has to be an arc as when used - it is shared between the metrics and actual service's logic to respond correctly on the probe.